### PR TITLE
Project maintenance

### PR DESCRIPTION
**Pull request by metart43**
## What's changed

This project will be migrated to TypeScript in near future. JavaScript files responsible for rendering typeahead and running demo will be removed or/and reworked. 

This PR sets up and configures linting, Typescript and removes `snyk`. I also ran formatting on all project files.

#### Note

`"demos/app.js", "index.js", "webpack.config.js"` were added to `eslint` ignore because they are breaking the `pre-commit` checks. This will be removed once we get rid the redundant JS files.

### CODE REVIEW

File: .toolkitrc.yml [hunk 1]
```diff
@@ -3,20 +3,28 @@ plugins:
...
-    files: "!(public)/**/*.js"
+    files:
+      - "**/*.{js,jsx,ts,tsx,yaml,yml,json}"
   "@dotcom-tool-kit/webpack":
     configPath: "webpack.config.js"
   "@dotcom-tool-kit/circleci":
     nodeVersion:
-      - '18.16'
-      - '16.20'
+      - "18.16"
```
Comment (kavanagh) [inline .toolkitrc.yml:29]: I'd go with 18.x, 20.x and 22.x. Don't bother with 16. People are going to soon start adopting v22 and we'll need to support that.

File: demos/app.js [hunk 2]
```diff
@@ -1,49 +1,48 @@
-'use strict';
+"use strict";
```
Comment (kavanagh) [inline demos/app.js:1]: This PR is using double quotes everywhere. Has Prettier been run on the whole project yet? That should use single quotes if we're using the same rules as search-page, next-home-page etc.


Also the project still has semicolons.

File: prettier.config.js
Comment (kavanagh) [inline prettier.config.js:1]: Where did the content of this file come from? Would be good to use the same as our other projects.

Also I think we're going to need an .editorconfig file.

Comment (metart43) [reply prettier.config.js:1]: I took it form the search-page project, bit deliberately changed the `singleQuote` to `false`. It's just my personal preference. I think it aligns better when `jsxSingleQuote` is `false`, but yeah you are right. We should have consistency across our projects.

File: demos/app.js (hunk 2 above)
Comment (metart43) [reply demos/app.js]: Done. Re-run the formatting after adding the `next-search-page` config

File: prettier.config.js
Comment (metart43) [reply prettier.config.js:1]: DOne

File: .toolkitrc.yml (hunk 1 above)
Comment (metart43) [reply .toolkitrc.yml]: Done here https://github.com/Financial-Times/n-topic-search/pull/64/commits/3c179296dd80660a9543fb0de55eae6e0054377e

**Review by kavanagh — APPROVED** [review]
LGTM. Although I'm not completely sure if this is safe to merge. Will let you decide that...

**Comment by metart43** [comment]
@kavanagh thank you for your review. I have just tested this locally with pagekit and homepage and merging this changes leave the project in deployable state. The typeahead still works as expected

**Pull request merged**